utils/fixed_decimal/src/decimal.rs

```diff
@@ -471,8 +471,14 @@ impl FixedDecimal {
                 return;
             }
             let _ = self.digits.drain(0..cut as usize).count();
-            self.magnitude = magnitude;
+            // Count number of leading zeroes
+            let extra_zeroes = self.digits.iter().position(|x| *x != 0).unwrap_or(0);
+            let _ = self.digits.drain(0..extra_zeroes).count();
+            self.magnitude = magnitude - extra_zeroes as i16;
             self.upper_magnitude = positive_magnitude;
+            if self.lower_magnitude > self.magnitude {
+                self.lower_magnitude = self.magnitude;
+            }
```

Comment:
Nit: I don't think this condition can be triggered here.

Reply:
I thought so too, but I figured it didn't hurt to add to avoid hitting broken invariants. But I think it's better to remove.